frontends/web/src/components/confirm/Confirm.tsx

```diff
@@ -14,85 +14,74 @@
  * limitations under the License.
  */
 
-import { Component} from 'react';
-import { translate, TranslateProps } from '../../decorators/translate';
+import { FunctionComponent, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
 import { SimpleMarkup } from '../../utils/markup';
 import { Dialog, DialogButtons } from '../dialog/dialog';
 import { Button } from '../forms';
 
-let confirmation: (message: string, callback: (response: boolean) => void, customButtonText?: string) => void;
+type TCallback = (response: boolean) => void;
+
+/**
+ * @deprecated
```

Comment:
if deprecated, there should be an alternative. what's the alternative?

Reply:
Just use a dialog in the comonent instead of calling the global instance.